### PR TITLE
Adding examples to the APIs to improve documentation

### PR DESCRIPTION
**Pull request by gucalder**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: Would you like to enforce a pattern here or does the server handle this.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: Setting modelAsString to false may result in breaking changes whenever the enum list is modified.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: Please make sure the PUT/DELETE operations here are not long running

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: nit: the good news is you can now cross reference between specs, so you can define ```Resource``` in ```actionGroups_API.json``` and reference it here, saves you the time and trouble replication can cost.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: Please expose a PATCH operation for the tracked resource ```ActionGroupResource```

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: Please expose a PATCH operation for the tracked resource ```ActivityLogAlertResource```

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: Should have a ```nextLink``` property

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: fixed

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: Fixed

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: Fixed

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: @wilyle has more context here.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: @wilyle has more context here.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: That's the value (i.e. false) we use in all the other enum declarations. Maybe because we rarely change these lists.
I am not sure if @wilyle can foresee changes in this particular list.

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: @davmc Do we have a PATCH implemented already?

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: Please enlighten us! :) 
I think we would need to use "$ref": "and then the relative path to the file and then #/definitions/Resources" to do this right?

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: Fixed

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: Thats right, an [example](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-recoveryservices/2016-06-01/swagger/vaultusages.json#L46) would serve you better though

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (wilyle) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: I do not think these are long-running. @georgenm can verify

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (wilyle) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: The patch operation only supports tags, so this was intentionally omitted according to the [swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md):

> M2056: If the service (in Azure) only supports PATCHING tags on a Resource in the PATCH operation and has a PUT on the same resource which is named as "CreateOrUpdate", then it MUST NOT expose the "PATCH" operation in the swagger spec.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (wilyle) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: We have changed these values in the past before the service was public, but to prevent breaking changes we have committed to leaving this list as-is

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (wilyle) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: We likely only support patching tags (I'm not sure as I'm not very familiar with the activity log alerts apis), in which case the swagger checklist in the documentation specifies that the PATCH operation should not be exposed (see my other comment on the same for `ActionGroupResource`)

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: If you're only supporting PATCH for tags, this should be fine.

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: If you're only supporting PATCH for tags, this should be fine. Please confirm the same.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (wilyle) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: the server is doing validation here

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: We have several places where I could use this cross reference. I might need to plan to make this change though. Can I leave this change for the future?
I also have a question, I am allowed to define a file with common declarations and put it in the root of my folder and reference it from there? What should the file contain?

Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: A. Yes, I won't block you on this one.
B. Yes, consolidating definitions in a common document sounds ideal. I'm not sure what you mean by root folder, as long as you can specify a relative path you should be fine. Having said that you might wish to hold off a little given some upcoming directory structure changes here.
C. The document can be just like any other swagger document but devoid of paths section but could contain almost of the common definitions.

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: I just added the PATCH operation, which was left out from the previous version, but should not be.

File: arm-monitor/2017-04-01/swagger/actionGroups_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/actionGroups_API.json]: I just confirmed with @georgenm. These operations are not long-running.

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (dsgouda) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: This really isn't a resource object, looks like a simple request parameter. Instead of an allOf on Resource, consider setting location as a required property (if needed)
The reason I suggest is because ```ActivityLogAlertResourcePatch``` will be otherwise considered as a resource itself by our validation rules.

**Review by dsgouda — COMMENTED** [review]
Looks good apart from a minor issue. 
Also, please make sure there exists an operations API that exposes information about the new features you are trying to expose here.

File: arm-monitor/2015-04-01/swagger/autoscale_API.json
Comment (dsgouda) [inline arm-monitor/2015-04-01/swagger/autoscale_API.json]: Whoops, nearly missed this! If hours can be a part of any request parameter, this could be a breaking change for your sdk. FYI

Comment (gucalder) [inline arm-monitor/2015-04-01/swagger/autoscale_API.json]: I am removing the minimum/maximum from hours and minutes, the server will reject any value outside those limits anyway.

Comment (gucalder) [inline arm-monitor/2015-04-01/swagger/autoscale_API.json]: I am removing the minimum/maximum from hours and minutes, the back-end rejects any value outside those limits anyway. It has always done that.

File: arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json
Comment (gucalder) [inline arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json]: I do not understand this one, sorry.
Location is already required in Resource.
The goal of this PATCH is to allow for changes in tags and in enabled properties.
We cannot use ActivityLogAlertResource here because it marks scopes, condition and actions as required, but the patch will ignore them.

**Review by dsgouda — APPROVED** [review]

**Comment by dsgouda** [comment]
Will merge once CI builds are happy

**Pull request merged**
